index.js

```diff
@@ -452,6 +455,21 @@ function trayCreate(json) {
     client.write(json.targetID, consts.eventNames.trayEventCreated)
 }
 
+// trayPopUpContextMenu pops up the context menu of the tray
+function trayPopUpContextMenu(json) {
+    let menu = menuCreate(json.menu);
+    if (!menu && !Object.keys(json.menuPopupOptions).length) {
+        elements[json.targetID].popUpContextMenu();
+    } else if (menu && !Object.keys(json.menuPopupOptions).length) {
+        elements[json.targetID].popUpContextMenu(menu);
+    } else if (!menu && Object.keys(json.menuPopupOptions).length) {
+        elements[json.targetID].popUpContextMenu(json.menuPopupOptions);
+    } else {
+        elements[json.targetID].popUpContextMenu(menu, json.menuPopupOptions);
+    }
+    client.write(json.targetID, consts.eventNames.trayEventPoppedUpContextMenu);
```

Comment:
Could you move the `client.write()` part outside of this function, in the big switch where `trayPopUpContextMenu()` is called?

Reply:
Done.